src/advertisementservice.cpp

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2017 - Florent Revest <revestflo@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <QDBusInterface>
+#include <QDBusMessage>
+
+#include "advertisementservice.h"
+#include "characteristic.h"
+#include "common.h"
+
+AdvertisementService::AdvertisementService(int index, QDBusConnection bus, QObject *parent) : Service(bus, index, ADVERT_UUID, parent)
+{
+    // A stub service
```

Comment:
Given that this file solely exists to define an empty constructor and will never evolve, you might as well inline the constructor definition within the class declaration and remove this cpp file